internal/tui/components/dialog/setup.go

```diff
@@ -0,0 +1,547 @@
+package dialog
+
+import (
+	"github.com/charmbracelet/bubbles/cursor"
+	"github.com/charmbracelet/bubbles/key"
+	"github.com/charmbracelet/bubbles/textinput"
+	tea "github.com/charmbracelet/bubbletea"
+	"github.com/charmbracelet/lipgloss"
+	"github.com/sst/opencode/internal/llm/models"
+	"github.com/sst/opencode/internal/tui/layout"
+	"github.com/sst/opencode/internal/tui/styles"
+	"github.com/sst/opencode/internal/tui/theme"
+	"github.com/sst/opencode/internal/tui/util"
+	"strings"
+)
+
+type SetupDialog interface {
+	tea.Model
+	layout.Bindings
+}
+
+// AvailableProviders returns a list of all available providers
+func AvailableProviders() ([]models.ModelProvider, map[models.ModelProvider]string) {
+	providerLabels := make(map[models.ModelProvider]string)
+	providerLabels[models.ProviderAnthropic] = "Anthropic"
+	providerLabels[models.ProviderAzure] = "Azure"
+	providerLabels[models.ProviderBedrock] = "Bedrock"
+	providerLabels[models.ProviderGemini] = "Gemini"
+	providerLabels[models.ProviderGROQ] = "Groq"
+	providerLabels[models.ProviderOpenAI] = "OpenAI"
+	providerLabels[models.ProviderOpenRouter] = "OpenRouter"
+	providerLabels[models.ProviderXAI] = "xAI"
```

Comment:
i think we need these friendly provider names/labels in the model switcher dialog as well, can you pull these out somewhere so that they can be leveraged there as well?

Reply:
Addressed via https://github.com/sst/opencode/pull/14/commits/455e5c1104d11d993135c420b7174d00b6e88fc9 and partly in https://github.com/sst/opencode/pull/14/commits/4e47ad68cd5632997d0af1652bab8209060ef696